torchtune/datasets/_prompt_classification.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Based off of TextCompletionDatset in _text_completion.py from torchtune
+
+from typing import Any, Dict, List, Mapping, Optional
+
+from datasets import load_dataset, load_from_disk
+from torch.utils.data import Dataset
+from torchtune.data import truncate
+from torchtune.datasets._packed import PackedDataset
+from torchtune.modules.tokenizers import ModelTokenizer
+
+
+class PromptClassificationDataset(Dataset):
+    """
+    Modified dataset from TextCompletionDataset for classifying any unstructured text corpus. Data format would be
+    in a prompt and completion format similar to openai gpt-3 finetuning. Quickly load any dataset
+    from Hugging Face or local disk and tokenize it for your model.
+
+    Args:
+        tokenizer (ModelTokenizer): Tokenizer used by the model that implements the ``tokenize_messages`` method.
+        source (str): path string of dataset, anything supported by Hugging Face's ``load_dataset``
+            (https://huggingface.co/docs/datasets/en/package_reference/loading_methods#datasets.load_dataset.path)
+        from_disk (bool): Whether to load the dataset from disk. Default is False.
+        text_column (str): Name of column in the sample that contains the text data. This is typically required
+        label_column (str): Name of column in the sample that contains the label data. Default is "label".
+        split (str): Name of the split to load from the dataset. Default is "train".
+        classes (Optional[List[Any]]): List of classes for label encoding. If None, the classes will be created from the labels
+            in the dataset. Default is None.
+        max_seq_len (Optional[int]): Maximum number of tokens in the returned input and label token id lists.
+            Default is None, disabling truncation. We recommend setting this to the highest you can fit in memory
+            and is supported by the model. For example, llama2-7B supports up to 4096 for sequence length.
+        **load_dataset_kwargs (Dict[str, Any]): additional keyword arguments to pass to ``load_dataset``.
+    """
+
+    def __init__(
+        self,
+        tokenizer: ModelTokenizer,
+        source: str,
+        from_disk: bool = False,
+        text_column: str = "text",
+        label_column: str = "label",
+        split: str = "train",
+        classes: Optional[List[Any]] = None,
```

Comment:
class/label mapping currently passed in as a an ordered list. Not sure what's the best practice for this (do we store a separate label mapping file somewhere?)